src/pages/index.js

```diff
@@ -0,0 +1,264 @@
+  confirmationPopup.open(); //opening the popup
+  confirmationPopup.setSubmitAction(() => {
+    confirmationPopup.setDeleteLoading(true); //cahnge the text content of the confirm button
+    api
+      .deleteCard(card._id)
+      .then((result) => {
+        card.removeCardElement(result); //card.js method to remove the card_element
+        if (result) {
+          confirmationPopup.close();
+        }
+      })
+      .catch((err) => {
+        console.error(`${err} Failed to delete post.`);
+        console.error("Error in deleteCard:", error);
+        return Promise.reject(error);
+      })
+      .finally(() => {
+        confirmationPopup.setDeleteLoading(false);
+      });
+  });
+}
+/*
+handleLikedButtonClick funtion will be called after the event click on the like button
+this function is passed to the card class
+at the begining its checking if the isLiked is true or false and by the state
+of the isLiked its decide's wich api method to call 
+after the api call we manualy changing the isLiked atribute of the img with the
+setLike function from the card class that adding or removing the like button
+active class
+*/
+function handleLikedButtonClick(card) {
+  if (card.isLiked) {
+    api
+      .dislikeCard(card._id)
+      .then((res) => {
+        card.setLike(res.isLiked);
+      })
+      .catch((error) => {
+        console.error("Error in dislikeCard:", error);
+        return Promise.reject(error);
+      });
+  } else {
+    api
+      .likeCard(card._id)
+      .then((res) => {
+        card.setLike(res.isLiked);
+      })
+      .catch((error) => {
+        console.error("Error in the LikeCard:", error);
+        return Promise.reject(error);
+      });
+  }
+}
+
+function upDateAvatar(avatar) {
+  avatarPopup.setLoading(true);
+  api
+    .upDateAvater(avatar.link)
+    .then((res) => {
+      avatarImage.src = res.avatar;
+      avatarPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error in upDateAvatar:", error);
+      return Promise.reject(error);
+    })
+    .finally(() => {
+      avatarPopup.setLoading(false);
+    });
+}
+
+///////////////////////////////////////////////////////// Event Listeners /////////////////////////////////////////////////////////////////
+
+//////////////////////////// modal events ///////////////////
+// editButton.addEventListener("click", function () {
+//   api
+//     .getUserInfo()
+//     .then((userData) => {
+//       userInfo.getUserInfo(userData.name, userData.about);
+//       editFormValidator.resetValidation();
+//       nameInput.value = userData.name;
+//       jobInput.value = userData.about;
+//       newEditPopup.open();
+//     })
+//     .catch((error) => {
+//       console.error("Error in getUserInfo:", error);
+//       return Promise.reject(error);
+//     });
+// });
+
+editButton.addEventListener("click", function () {
+  nameInput.value = profileName.textContent;
+  jobInput.value = profileJob.textContent;
```

Comment:
`textContent` shouldn't be used in `index.js` anymore. You have `UserInfo` for getting the user info

Reply:
i fixed it